packages/@aws-cdk/aws-appsync/lib/data-source.ts

```diff
@@ -81,13 +103,41 @@ export interface ExtendedDataSourceProps {
 }
 
 /**
- * Abstract AppSync datasource implementation. Do not use directly but use subclasses for concrete datasources
+ * An Interface for Data Sources
  */
-export abstract class BaseDataSource extends Construct {
+export interface IDataSource extends IConstruct {
+  /**
+   * the name of the data source
+   */
+  readonly name: string;
+  /**
+   * the arn of the data source
+   */
+  readonly arn: string;
+  /**
+   * the underlying CFN data source resource
+   */
+  readonly ds: CfnDataSource;
+  /**
+   * creates a new resolver for this datasource and API using the given properties
+   */
+  createResolver(props: BaseResolverProps): Resolver;
```

Comment:
we tend to name these API `add<blah>` - any reason why we would not want to call this `addResolver` ?

Reply:
it was like that before so i didnt want to introduce more breaking changes.. but if we feel strongly to change it then i can change it in this PR.. same with the `BaseDataSource` comment